tokio/src/runtime/shell.rs

```diff
@@ -0,0 +1,79 @@
+use crate::runtime::time;
+use crate::runtime::{enter, io, Park};
+
+use std::future::Future;
+use std::mem::ManuallyDrop;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::task::Poll::Ready;
+use std::task::{Context, RawWaker, RawWakerVTable, Waker};
+
+#[derive(Debug)]
+pub(super) struct Shell {
+    driver: time::Driver,
```

Comment:
The `Shell` runtime supports using Tokio types w/o spawning a full runtime.

`time::Driver` is an alias for `tokio::time::Driver<tokio::net::driver::Driver>`, depending on conditional compilation.